Firmware/MotorControl/thermistor.hpp

```diff
@@ -0,0 +1,132 @@
+    ThermistorCurrentLimiter(ThermistorCurrentLimiter&& other) noexcept = delete;
+    ThermistorCurrentLimiter& operator=(const ThermistorCurrentLimiter& original) = delete;
+    ThermistorCurrentLimiter& operator=(ThermistorCurrentLimiter&& original) = delete;
+
+    ThermistorCurrentLimiter(uint16_t adc_channel,
+                             const float* const coefficients,
+                             size_t num_coeffs,
+                             const float& temp_limit_lower,
+                             const float& temp_limit_upper,
+                             const bool& enabled);
+
+    float get_temp() const;
+
+    void update();
+
+    bool do_checks();
+
+    float get_current_limit(float user_limit) const override;
+
+    uint16_t adc_channel_;
+    const float* const coefficients_;
+    const size_t num_coeffs_;
+    float temp_;
+    const float& temp_limit_lower_;
+    const float& temp_limit_upper_;
+    const bool& enabled_;
+    Error_t error_;
+    Axis* axis_ = nullptr; // set by Axis constructor
+};
+
+DEFINE_ENUM_FLAG_OPERATORS(ThermistorCurrentLimiter::Error_t)
+
+class OnboardThermistorCurrentLimiter : public ThermistorCurrentLimiter {
+public:
+    struct Config_t {
+        float temp_limit_lower = 100;
+        float temp_limit_upper = 120;
+    };
+
+    // Communication protocol definitions
+    auto make_protocol_definitions() {
+        return make_protocol_member_list(
+            make_protocol_property("error", &error_),
+            make_protocol_ro_property("temp", &temp_),
+            make_protocol_object("config",
+                make_protocol_property("temp_limit_lower", &config_.temp_limit_lower),
+                make_protocol_property("temp_limit_upper", &config_.temp_limit_upper)
+            )
+        );
+    }
+
+    OnboardThermistorCurrentLimiter() = delete;
+    virtual ~OnboardThermistorCurrentLimiter() = default;
+    OnboardThermistorCurrentLimiter(const OnboardThermistorCurrentLimiter& other) = delete;
+    OnboardThermistorCurrentLimiter(OnboardThermistorCurrentLimiter&& other) noexcept = delete;
+    OnboardThermistorCurrentLimiter& operator=(const OnboardThermistorCurrentLimiter& original) = delete;
+    OnboardThermistorCurrentLimiter& operator=(OnboardThermistorCurrentLimiter&& original) = delete;
+
+    OnboardThermistorCurrentLimiter(const ThermistorHardwareConfig_t& hw_config, Config_t& config);
+
+    Config_t config_;
+
+private:
+    // Singleton true to pass as reference to ThermistorCurrentLimiter
+    // as enable.
+    static const bool true_singleton;
+};
+
+class OffboardThermistorCurrentLimiter : public ThermistorCurrentLimiter {
+public:
+    static const size_t num_coeffs_ = 4;
+
+    struct Config_t {
+        float thermistor_poly_coeffs[num_coeffs_] = { };
+        uint16_t gpio_pin = 4;
+        float temp_limit_lower = 100;
+        float temp_limit_upper = 120;
+        bool enabled = false;
```

Comment:
Technically a thermistor can just be disabled by putting the limits to infinity. On the other hand there is some merit to having an explicit enable flag. In any case we should make it consistent between onboard and offboard thermistors.

Reply:
I didn't include an enable for onboard thermistors because I didn't think we should support disabling them. An enable flag is definitely needed for the offboard thermistors because else we possibly sample unconnected IO and create bogus temps that then trigger the temperature limits.

Reply:
Fixed. Added enable to onboard thermistors as well.